app/views/shared/_run_table.slim

```diff
@@ -29,7 +30,17 @@
             - if cols.include?(:runner)
               td.align-right = user_badge(run.user)
             - if cols.include?(:time)
-              td.align-right = run.duration.format
+              td.align-right
+                - if cols.include?(:archived) && run.category_id.present?
+                  - old_runs = all_runs.filter { |r| r.id != run.id && r.category_id == run.category_id }
```

Comment:
Is the first part of the filter, where we avoid displaying the run itself, moveable to the `all_runs` query up above? Seems like these runs are getting pulled from the database into memory twice, and the second time is being ignored.